boa_engine/src/lib.rs

```diff
@@ -123,9 +123,11 @@
 
     // Add temporarily - Needs addressing
     clippy::missing_panics_doc,
-    clippy::arc_with_non_send_sync,
 )]
 
+#[cfg(not(target_has_atomic = "ptr"))]
+compile_error!("Boa requires a lock free `AtomicUsize` in order to work properly.");
+
```

Comment:
We don't need this, right?

Reply:
I see, forgot that `Arc` uses `AtomicUsize` internally 😅 . thanks for the clarification!